bioio_sldy/sldy_image.py

```diff
@@ -219,6 +221,14 @@ def __init__(
         self.timepoints = sorted(self._timepoint_to_data_paths.keys())
         self.channels = sorted(self._channel_to_data_paths.keys())
 
+        self.sizeT = self._image_record["CImageRecord70"]["mNumTimepoints"]
+        self.sizeC = self._image_record["CImageRecord70"]["mNumChannels"]
+        self.sizeZ = self._image_record["CImageRecord70"]["mNumPlanes"]
+        self.sizeY = self._image_record["CImageRecord70"]["mHeight"]
+        self.sizeX = self._image_record["CImageRecord70"]["mWidth"]
+        # TODO check this but are all sldys uint16?  bioformats seems to say so
+        self.dtype = np.dtype(np.uint16)
```

Comment:
Quick check: have we found any files that have pixel data <-> metadata conflicts?

Reply:
I assume you are putting these values into their own attributes because we don't have a system in place for storing pre-read / pre-load dims?

Reply:
Right, I made a ticket in bioio-base about this.  Even if the xarray delayed setup is "fast", it would be still faster to return dims from metadata.

We had czi files that had pixel data size different from metadata size but as I recall it was generally because of unexpected termination of the image capture.

Reply:
I think we should just move to standardize this sooner rather than later and create a `dims_from_meta` attribute or something that Readers can optionally impl.

The docs for the attr should just mention that we have no guarantee that the dims are correct due to "early stopping of image acquisition"